File: mnist-client/src/pages/api/predict.js
import axios from 'axios';

export default async function handler(req, res) {
    if(req.method === 'POST') {
        const response = await axios.post('http://localhost:8085/predict', req.body);
        const { data: {
            data
        }} = response;
        const [prediction  = []] = data;
        prediction.forEach((r) => r * 100)
        const biggestIndex = prediction.indexOf(Math.max(...prediction));
        res.status(200).json({ mightBe: biggestIndex });
        return;
    }
    res.status(400).json({error: 'Method not allowed'})
}

File: predict.R
# call the model and predict the number
#* Process data received in a POST request
#* @post /predict
function(req, res) {
  # Access the request body
  body <- req$body
 
  prediction <- model %>% predict(array_reshape(body$value, c(1, 28*28)))
  print(prediction)
  # # Return a response
  response <- list(data = prediction)

  # # Set the response status code (optional)
  res$status <- 201 # Created

  # # Return the response as JSON
  return(response)
}
